src/components/task-list/task-list.component.js

```diff
@@ -1,26 +1,42 @@
-import { useState } from "react";
 import styles from "./task-list.module.css";
 
 const TaskList = (props) => {
-  const [completed, setCompleted] = useState([]);
-  const tasks = props.tasks;
+  const { tasks, taskCompletionToggle, handleTaskCompleted } = props;
 
-  const toggleCompletion = (event) => {
-    setCompleted({ ...completed });
+  const completionToggle = (event) => {
+    taskCompletionToggle(event);
+  };
+
+  const clickCheckbox = (clickEvent) => {
+    handleTaskCompleted(clickEvent);
```

Comment:
passing the event "up" is "giving to the parent" too much responsibility... the parent now needs to know that the event is of a certain type and on and all it's too much for it to handle... it'll be better to extract the id directly here and just pass it to `handleTaskCompleted`. Something in between these lines:

```typescript
    const taskId = +clickEvent.target.parentElement.attributes.id.value;
    handleTaskCompleted(clickEvent);
```

Also, I would suggest moving the id to the `checkbox` (so you don't have to access to its parent in the handler to retrieve the value)

Reply:
7587e2b3c962486f4d1bd4d9cae5891e5f7c0097 Changed as advised.